### PR TITLE
Add `ansible list` command

### PR DESCRIPTION
**Pull request by sterchelen**
- use aini https://github.com/relex/aini as vendor pkg
- multi inventories reader

ref https://github.com/peopledoc/peopleinfra/issues/8099

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -44,6 +44,7 @@ func init() {
 		StringVarP(&envName, "env", "e", "", "Environment name, syntax(type.env.platform)")
 	rootCmd.PersistentFlags().
 		BoolVar(&isDebug, "debug", false, "debug mode")
+	rootCmd.MarkPersistentFlagRequired("env")
```
Comment (SwannCroiset) [inline cmd/root.go:47]: env is now mandatory, any reason ?

File: cmd/ansible.go [hunk 2]
```diff
@@ -11,39 +13,63 @@ import (
...
 	ansibleCmd.AddCommand(runCmd)
+
+	//RunCmd
 	runCmd.Flags().StringVarP(&ModuleName, "module", "m", "shell", "Ansible module name")
 	runCmd.Flags().StringVarP(&ModuleArg, "args", "a", "", "Ansible module arg")
 	runCmd.Flags().StringVarP(&HostPattern, "target", "t", "", "Ansible host-pattern")
 	runCmd.Flags().BoolVar(&CheckModeEnabled, "check", false,
 		"Enable check mode")
 	runCmd.MarkFlagRequired("target")
+	ansibleCmd.AddCommand(runCmd)
```
Comment (SwannCroiset) [inline cmd/ansible.go:65]: added twice `	ansibleCmd.AddCommand(runCmd)`

File: cmd/root.go (hunk 1 above)
Comment (sterchelen) [reply cmd/root.go:47]: It is needed for all our commands. It allows us to skip adding a `PreRunE` and check if `env` is filled :)

File: cmd/ansible.go (hunk 2 above)
Comment (sterchelen) [reply cmd/ansible.go:65]: Whoops :/ , thanks

Comment (sterchelen) [reply cmd/ansible.go:65]: Fixed.

File: cmd/ansible.go [hunk 3]
```diff
@@ -148,3 +173,70 @@ var playCmd = &cobra.Command{
...
+
+		allInventories, err := environment.GetFullPathInventoriesFromEnvironments(envsPath, *environments)
+		if err != nil {
+			return err
+		}
+
+		var invReaders []io.Reader
+		for _, invs := range allInventories {
+			for _, inv := range invs {
+				f, err := os.Open(inv)
```
Comment (SwannCroiset) [inline cmd/ansible.go:206]: need to check if "inv" is a readable file (and not a directory) b/c it fails silently with directory

File: cmd/ansible.go [hunk 4]
```diff
@@ -148,3 +173,70 @@ var playCmd = &cobra.Command{
...
+				if err != nil {
+					return err
+				}
+				invReaders = append(invReaders, bufio.NewReader(f))
```
Comment (SwannCroiset) [inline cmd/ansible.go:210]: in debug mode, I'd print the  loaded inventory files

File: cmd/ansible.go [hunk 5]
```diff
@@ -148,3 +173,70 @@ var playCmd = &cobra.Command{
...
+		r := io.MultiReader(invReaders...)
+		manipulator := ansible.InitInventoryManipulator(r)
+
+		if ListGroup {
+			groups, err := manipulator.GetGroupsName(WithParent)
+			if err != nil {
+				return err
+			}
+			//I know it is a bit odd but Jarvis is learning
+			//be nice with him :)
```
Comment (SwannCroiset) [inline cmd/ansible.go:227]: be bold :)

File: cmd/root.go (hunk 1 above)
Comment (SwannCroiset) [reply cmd/root.go:47]: if then env is not provided by default jarvis used to iterate over all envs no ?

**Review by SwannCroiset — COMMENTED** [review]
works for me :+1: 

some question/remarks in-line

File: cmd/ansible.go (hunk 3 above)
Comment (sterchelen) [reply cmd/ansible.go:206]: You are absolutely right :) ... adding this

File: cmd/ansible.go (hunk 4 above)
Comment (sterchelen) [reply cmd/ansible.go:210]: yes, I'll add it

File: cmd/root.go (hunk 1 above)
Comment (sterchelen) [reply cmd/root.go:47]: Yes you are right. But it is not documented and not sure if it is the good way to do it.
I don't see the use case. Do you see one? I think if we have one, we must be explicit on that case.
`--env all` ?

Comment (SwannCroiset) [reply cmd/root.go:47]: I see one usecase for `ansible run`, it's handy to run a command over all envs.

I like the idea to be explicit `--env all`

Comment (sterchelen) [reply cmd/root.go:47]: I propose to let the `MarkRequired...` for this PR and add the `--env all` as an issue?

Comment (SwannCroiset) [reply cmd/root.go:47]: k

**Review by SwannCroiset — APPROVED** [review]

**Pull request merged**